feat: packaging + introduce cli

Niceadam opened the pull request:
## Description
I propose introducting Aegra CLI tool for easier usage of the library. The PR mostly sets up the boilerplate but its functional in calling the existing server and migrate script. For additional changes, I would like to ask the maintainer for what direction he would like to take this in, what to change and how to expand it. 

Usage:
```bash
uv sync
source .venv/bin/activate
aegra --help
```

My idea is that this will replace the LangGraph CLI in existing projects, including commands like dockerfile generation and image building.

## Type of Change
<!-- Check the relevant option -->
- [x ] `feat`: New feature
- [ ] `fix`: Bug fix
- [ ] `docs`: Documentation changes
- [ ] `style`: Code style/formatting
- [ ] `refactor`: Code refactoring
- [ ] `perf`: Performance improvement
- [ ] `test`: Tests added/updated
- [ ] `chore`: Maintenance (dependencies, build, etc.)
- [ ] `ci`: CI/CD changes

## Changes Made
<!-- List the main changes -->
- cli tool written in [Click](https://click.palletsprojects.com/en/stable/)
- migrate migrate.py & run_server.py to be used by tool instead
- new cli commit scope 

## Testing
<!-- Describe how you tested your changes -->
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] E2E tests added/updated
- [x ] Manual testing performed

## Checklist
<!-- Ensure all items are checked before requesting review -->
- [ x] Code follows project style (Ruff formatting)
- [ x] All linting checks pass (`make lint`)
- [ ] Type checking passes (`make type-check`)
- [ ] All tests pass (`make test`)
- [ ] Documentation updated (if needed)
- [ x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)
- [x ] PR title follows Conventional Commits format

ibbybuilds commented:
@Niceadam good step. My vision for the CLI is this:

Python package integration: Aegra should be a Python package you can install directly in a LangGraph project to make it server-ready (similar to langgraph-api).

Local + deploy flow: The CLI should handle local serving, Docker image creation, and deployment, so users don’t have to manually place graphs into the repo’s /graphs folder as the current method requires.

Migration tooling: It should also support migrating from LangGraph Platform setups to Aegra setups for smoother adoption.

This would make Aegra the runtime layer that bridges LangGraph projects and production environments seamlessly.

Niceadam commented:
@ibbybuilds I am ready for another round of feedback, I apologize for introducing so many changes but it was necessary to get the repo ready for packaging.

Changes:
- `aegra` package is what will be shipped to users
- `aegra` now contains `agent_server` and `cli`
- alembic migrations are moved to `cli` to be packaged
- moved to using alembic python api
- `graphs` and `tests` are under src now, they don't get packaged
- fixed import/path stuff

Result:
You can now do this in your existing langgraph project
```bash
uv add /path/to/aegra
source .venv/bin/activate
aegra dev
```
It spins up a temporary postgres container, runs the migrations and starts the server and it should work as expected assuming your agents are somewhere under your `src` and it should work
